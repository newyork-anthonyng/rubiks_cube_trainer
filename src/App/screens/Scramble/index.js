import React, { Component } from "react";
import PropTypes from "prop-types";
import ScrambleDisplay from "../../shared/ScrambleDisplay";
import scrambler from "cube-scrambler";

class Scramble extends Component {
  constructor(props) {
    super(props);

    this.scrambler = scrambler();

    this.state = {
      scramble: props.initialScramble || this.getScramble()
    };
  }

  getScramble = () => {
    return this.scrambler.scramble().join(" ");
  };

  handleClick = () => {
    this.setState({
      scramble: this.getScramble()
    });
  };

  render() {
    return (
      <div>
        <ScrambleDisplay text={this.state.scramble} />
        <button onClick={this.handleClick}>Next Scramble</button>
      </div>
    );
  }
}

Scramble.propTypes = {
  initialScramble: PropTypes.string
};

export default Scramble;
