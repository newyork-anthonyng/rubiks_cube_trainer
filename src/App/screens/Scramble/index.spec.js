const mockScramble = jest.fn(() => ["U", "D2", "L", "R"]);
jest.mock("cube-scrambler", () => {
  return () => ({
    scramble: mockScramble
  });
});

import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";
import Scramble from "./";

beforeEach(() => {
  mockScramble.mockClear();
});

it("should render correctly", () => {
  const wrapper = shallow(<Scramble />);

  expect(toJSON(wrapper)).toMatchSnapshot();
});

it("should get new scramble when clicking on button", () => {
  const wrapper = shallow(<Scramble />);
  mockScramble.mockClear();

  const buttonEl = wrapper.find("button");
  buttonEl.simulate("click");

  expect(mockScramble).toHaveBeenCalledTimes(1);
});

it("should use initial scramble if provided", () => {
  const wrapper = shallow(<Scramble initialScramble="U R" />);

  expect(mockScramble).not.toHaveBeenCalled();
  expect(toJSON(wrapper)).toMatchSnapshot();
});
