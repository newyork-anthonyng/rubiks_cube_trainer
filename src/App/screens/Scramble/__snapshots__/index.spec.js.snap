// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div>
  <ScrambleDisplay
    text="U D2 L R"
  />
  <button
    onClick={[Function]}
  >
    Next Scramble
  </button>
</div>
`;

exports[`should use initial scramble if provided 1`] = `
<div>
  <ScrambleDisplay
    text="U R"
  />
  <button
    onClick={[Function]}
  >
    Next Scramble
  </button>
</div>
`;
